src/mbgl/map/map.cpp

```diff
@@ -899,10 +899,15 @@ std::unique_ptr<Layer> Map::removeLayer(const std::string& id) {
     impl->styleMutated = true;
     BackendScope guard(impl->backend);
 
-    auto removedLayer = impl->style->removeLayer(id);
-    impl->onUpdate(Update::Classes);
-
-    return removedLayer;
+    try {
+        auto removedLayer = impl->style->removeLayer(id);
+        impl->onUpdate(Update::Classes);
+        return removedLayer;
+    } catch (const std::runtime_error& err) {
+        impl->styleMutated = false;
```

Comment:
I believe this is incorrect. It always sets to false, even when the style was mutated before the call to remove a layer.